2 new Nodes that deal with TextureArrays

sebllll opened the pull request:
Hello Mr. Vux,

These 2 Nodes can come quite handy when dealing with Texturearrays.

**GetSlice (DX11.TextureArray ToArray)** takes a Spread of Textures and creates a new Array for each Index Bin.
**GetArray (DX11.TextureArray)** does quite the Same but for TextureArrays and not (yet) binsizeable.

I'm not sure if you have some improvements or see fundamental bugs, so consider this a proposal where I'm open to discuss things widely. So, any suggestions or critics are super welcome!

Helppatches are available (see images), but since all the helppatches are in the girlpower repo, i can add them if the PR is getting merged.

![image](https://user-images.githubusercontent.com/646501/38882133-b6eafe24-4269-11e8-818c-4e92ca07bc95.png)

![image 1](https://user-images.githubusercontent.com/646501/38882141-bd6520b8-4269-11e8-9379-d4930c1a5417.png)

mrvux commented:
Hey thanks for the pr, could you put the 2 help patches in the girlpower repo (otherwise it will make the integration tests fail), then I'll also merge this one

sebllll commented:
super

i've now added some refinements and gave the nodes better names.

the appveyor build still fails, though i've added the helppatches. maybe the build system is confused, because the referenced girlpower is my fork... 
anyway, the appveyor log is kinda cryptic... looks like there's also some verification error or **$env:priv_key** is not accessible

sebllll commented:
this is the log:
```
Build started
git clone -q https://github.com/mrvux/dx11-vvvv.git C:\projects\dx11-vvvv
git fetch -q origin +refs/pull/340/merge:
git checkout -qf FETCH_HEAD
Running Install scripts
$fileContent = "-----BEGIN RSA PRIVATE KEY-----`n"
$fileContent += $env:priv_key.Replace(' ', "`n")
You cannot call a method on a null-valued expression.
At line:1 char:1
+ $fileContent += $env:priv_key.Replace(' ', "`n")
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
 
Command executed with exception: You cannot call a method on a null-valued expression.
```

mrvux commented:
Yes don't worry about that for now, indeed submodules are a bit annoying, I normally do a run myself so it's fine

mrvux commented:
This is now merged, thanks!

mrvux commented:
As a side note, I added your help files myself manually, since somehow pull request with some girlpower submodule fork was bit messy.
